app/assets/scripts/components/documents/single-edit/index.js

```diff
@@ -249,3 +268,33 @@ const SaveButton = () => {
     </Tip>
   );
 };
+
+const SaveAndContinueButton = ({ nextStep }) => {
+  const history = useHistory();
+  const { dirty, isSubmitting, submitForm, status } = useFormikContext();
+  const submitAndContinue = useCallback(
+    (e) => {
+      submitForm(e).then(({ error, data }) => {
+        if (!error && nextStep) {
+          history.replace(documentEdit(data.alias, data.version, nextStep));
+        }
+      });
+    },
+    [submitForm, history, nextStep]
+  );
+
+  return (
+    <ButtonSecondary
+      title='Save current changes and continue to next step'
+      disabled={isSubmitting || !dirty || status?.working}
+      onClick={submitAndContinue}
+      useIcon='tick--small'
+    >
+      Save {nextStep && ' and continue'}
```

Comment:
I was wondering whether we should have a "Save and view" button on the last step to redirect the user to the view page. thoughts?

Reply:
My suggestion is to use "Save & continue" instead of "Save and continue". And "Save & view" on the last step.